src/sexpr.rs

```diff
@@ -393,6 +406,10 @@ impl<'a> Lexer<'a> {
             break;
         }
 
+        // NOTE(rachitnigam): Not sure if this is the best way to signal an
+        // error in the lexer.
+        assert!(!quoted, "unterminated | in symbol");
```

Comment:
Good point! We didn't really have any parse errors before, given how permissive the symbol parsing was, so it would be nice to revisit that in a follow-up PR.

Reply:
Yeah, some simple API to report the errors would be useful!